### DISCUSSION TITLE
Retrieve remote data from items

### DISCUSSION THREAD

**Discussion by adamDilger** [orchidjs/tom-select#341] [Ideas]
After loading and rendering remote data, there doesn't seem to be an easy way to get access to the underlying objects returned from the request. Events `onChange`, `onItemAdd` and methods `getItem`, `getOption` etc all just return either the value or the dom elements.

Could one of these existing events also return the underlying data object? e.g as an idea:

```typescript
// tom-select.ts
addItem( value:string, silent?:boolean ):void{
   ...
   // self.trigger('item_add', hashed, item);  <-- no data returned

   const { $id, $div, $order, $option, ...data } = self.options[hashed];  // ignore extra $___ fields
   self.trigger('item_add', hashed, item, data);
   ...
```

Unless this data is kept private for a certain reason, or I've missed how to access this data in the docs.

Thanks! :)

**Comment by awh-tokyo** [comment]
The options array can be accessed from your callback though:

```
   this.tomSelect = new TomSelect(this.dropdownTarget as TomInput, {
    ...
    onChange: this.valueChanged,
    ...
    } as unknown as TomSettings)


   valueChanged(value): void {
        console.log(value);

        const item = this.options[value];
        console.log(item);
  }
```
